writter.go

```diff
@@ -0,0 +1,23 @@
+package isatty
+
+import "io"
+
+// IsWriterTerminal return true if the writer is terminal.
+func IsWriterTerminal(w io.Writer) bool {
+	if f, ok := w.(filelike); ok {
+		return IsTerminal(f.Fd())
+	}
+	return false
+}
+
+// IsWriterCygwinTerminal return true if the writer is a cygwin or msys2 terminal.
+func IsWriterCygwinTerminal(w io.Writer) bool {
+	if f, ok := w.(filelike); ok {
+		return IsCygwinTerminal(f.Fd())
+	}
+	return false
+}
+
+type filelike interface {
```

Comment:
Use a type alias to avoid defining a real type:

```suggestion
type filelike = interface {
```

Reply:
Definitely can do it. I'm just wondering why? The type is private, so what's the point?